karapace/schema_reader.py

```diff
@@ -11,15 +11,61 @@
 from karapace.schema_models import SchemaType, TypedSchema
 from karapace.statsd import StatsClient
 from karapace.utils import KarapaceKafkaClient
-from queue import Queue
-from threading import Lock, Thread
+from threading import Event, Lock, Thread
 from typing import Dict
 
 import logging
 import time
 import ujson
 
 log = logging.getLogger(__name__)
+Offset = int
+
+# The value `0` is a valid offset and it repreesents the first message produced
+# to a topic, therefore it can not be used.
+OFFSET_EMPTY = -1
```

Comment:
I hit this bug while implementing the alternative https://github.com/aiven/karapace/pull/373. Basically the code thought the first message was consumed already because the start value was incorrect, so the follow up request didn't get the produced schema and the test failed.